src/presets/vercel.ts

```diff
@@ -173,10 +173,12 @@ function generateBuildConfig(nitro: Nitro) {
           src: key.replace(/^(.*)\/\*\*/, "(?<url>$1/.*)"),
           dest: generateEndpoint(key) + "?url=$url",
         })),
-      {
-        src: "/(.*)",
-        dest: "/__nitro",
-      },
+      ...(!nitro.options.routeRules['/**']?.cache || !(nitro.options.routeRules['/**'].cache.swr || nitro.options.routeRules['/**']?.cache.static)
```

Comment:
Can you please explain it better in line above? Do we need to also handle other route rules for catch-all such as headers?